src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -43,41 +61,60 @@ const NoTokenDisplay = styled.div`
   padding: 0px 28px;
   gap: 8px;
 `
-const TokenRowsContainer = styled.div`
-  width: 100%;
-`
 
-function useFilteredTokens(tokens: TokenData[] | undefined) {
-  const filterString = useAtomValue(filterStringAtom)
+const DEFAULT_FILTER_STRING = ''
+const DEFAULT_SHOW_FAVORITES = false
+
+const hasNonDefaultFilters = (filterString: string, showFavorites: boolean) =>
+  filterString !== DEFAULT_FILTER_STRING || showFavorites !== DEFAULT_SHOW_FAVORITES
+
+function useFilteredTokens(
+  tokenData: TokenData[],
+  filterString: string,
+  showFavorites: boolean,
+  nonDefaultFiltersExist: boolean
+) {
   const favoriteTokenAddresses = useAtomValue(favoritesAtom)
-  const showFavorites = useAtomValue(showFavoritesAtom)
   const shownTokens =
-    showFavorites && tokens ? tokens.filter((token) => favoriteTokenAddresses.includes(token.address)) : tokens
+    showFavorites && tokenData ? tokenData.filter((token) => favoriteTokenAddresses.includes(token.address)) : tokenData
```

Comment:
the type signature `useFilteredTokens` says this check shouldn't be necessary and that the `.filter` call will always happen. Is that incorrect? should there be an `undefined` or `?` on the `tokenData` type?